apps/wallet/src/ui/app/pages/swap/utils.ts

```diff
@@ -72,113 +86,149 @@ export function getUSDCurrency(amount?: number | null) {
 	});
 }
 
-export async function isExceedingSlippageTolerance({
-	slipPercentage,
-	poolId,
-	deepBookClient,
-	conversionRate,
-	isAsk,
-	average,
-}: {
-	slipPercentage: string;
-	poolId: string;
-	deepBookClient: DeepBookClient;
-	conversionRate: number;
-	isAsk: boolean;
-	average: string;
-}) {
-	const convertedAverage = new BigNumber(average).shiftedBy(conversionRate).toString();
-
-	const { bestBidPrice, bestAskPrice } = await deepBookClient.getMarketPrice(poolId);
-
-	if (!bestBidPrice || !bestAskPrice) {
-		return false;
+export const maxSlippageFormSchema = z.object({
+	allowedMaxSlippagePercentage: z
+		.number({
+			coerce: true,
+			invalid_type_error: 'Input must be number only',
+		})
+		.positive()
+		.max(100, 'Value must be between 0 and 100'),
+});
+
+export function useCoinTypesFromRouteParams() {
+	const [searchParams] = useSearchParams();
+	const fromCoinType = searchParams.get('type');
+	const toCoinType = searchParams.get('toType');
+
+	// Both are already defined, just use them:
+	if (fromCoinType && toCoinType) {
+		return { fromCoinType, toCoinType };
 	}
 
-	const slip = new BigNumber(isAsk ? bestBidPrice.toString() : bestAskPrice.toString()).dividedBy(
-		convertedAverage,
-	);
+	// Neither is set, default to SUI -> USDC
+	if (!fromCoinType && !toCoinType) {
+		return { fromCoinType: SUI_TYPE_ARG, toCoinType: USDC_TYPE_ARG };
+	}
 
-	return new BigNumber('1').minus(slip).abs().isGreaterThan(slipPercentage);
+	return { fromCoinType, toCoinType };
 }
 
-function getCoinsFromBalanceChanges(coinType: string, balanceChanges: BalanceChange[]) {
-	return balanceChanges
-		.filter((balance) => {
-			return balance.coinType === coinType;
-		})
-		.sort((a, b) => {
-			const aAmount = new BigNumber(a.amount).abs();
-			const bAmount = new BigNumber(b.amount).abs();
+export function useGetBalance({ coinType, owner }: { coinType?: string; owner?: string }) {
```

Comment:
I thought we have this hook

Reply:
this 1 has the `select`, and its used in 2 different places so i abstract to a hook